src/overlays/actors/ovl_En_Dragon/z_en_dragon.c

```diff
@@ -37,130 +72,777 @@ const ActorInit En_Dragon_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dragon/func_80B5EF88.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dragon/func_80B5EFD0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dragon/func_80B5F3A4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dragon/func_80B5F418.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dragon/func_80B5F508.s")
+void EnDragon_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDragon* this = THIS;
+
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &gDeepPythonSkel, &gDeepPythonSmallSideSwayAnim, this->jointTable,
+                       this->morphTable, DEEP_PYTHON_LIMB_MAX);
+
+    this->actor.colChkInfo.health = 4;
+    this->actor.colChkInfo.damageTable = &sDamageTable;
+    this->actor.targetMode = 0xA;
+    Collider_InitAndSetJntSph(globalCtx, &this->collider, &this->actor, &sJntSphInit, this->colliderElements);
+
+    this->collider.elements[0].dim.scale = this->collider.elements[1].dim.scale = this->collider.elements[2].dim.scale =
+        this->collider.elements[3].dim.scale = this->collider.elements[4].dim.scale =
+            this->collider.elements[5].dim.scale = this->collider.elements[6].dim.scale =
+                this->collider.elements[7].dim.scale = 1.0f;
+
+    this->collider.elements[0].dim.modelSphere.radius = 150;
+    this->collider.elements[0].dim.modelSphere.center.x = 420;
+    this->collider.elements[1].dim.modelSphere.radius = 160;
+    this->collider.elements[1].dim.modelSphere.center.x = 630;
+    this->collider.elements[2].dim.modelSphere.radius = 130;
+    this->collider.elements[2].dim.modelSphere.center.x = 630;
+    this->collider.elements[3].dim.modelSphere.radius = 170;
+    this->collider.elements[3].dim.modelSphere.center.x = 920;
+    this->collider.elements[4].dim.modelSphere.radius = 150;
+    this->collider.elements[4].dim.modelSphere.center.x = 530;
+    this->collider.elements[5].dim.modelSphere.radius = 140;
+    this->collider.elements[5].dim.modelSphere.center.x = 730;
+    this->collider.elements[6].dim.modelSphere.radius = 120;
+    this->collider.elements[6].dim.modelSphere.center.x = 430;
+    this->collider.elements[7].dim.modelSphere.radius = 110;
+    this->collider.elements[7].dim.modelSphere.center.x = 160;
+
+    this->pythonIndex = EN_DRAGON_GET_PYTHON_INDEX(&this->actor);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->action = DEEP_PYTHON_ACTION_IDLE;
+    this->actor.hintId = 0xE;
+    this->scale = 0.5f;
+    this->actor.flags &= ~ACTOR_FLAG_8000000;
+
+    EnDragon_SetupRetreatOrIdle(this);
+}
+
+void EnDragon_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDragon* this = THIS;
+
+    Collider_DestroyJntSph(globalCtx, &this->collider);
+}
+
+AnimationHeader* animations[] = {
+    &gDeepPythonSmallSideSwayAnim,
+    &gDeepPythonLargeSideSwayAnim,
+    &gDeepPythonVerticalSwayAnim,
+    &gDeepPythonSmallSideSwayAnim,
+};
```

Comment:
You should either use a more unique name for named variables to avoid possibly future problems with repeated names or use the `static` keyword.
If you want to keep this name, you should use something like `sAnimations`.
This comment applies for the rest of the data of this actor